qldbdriver/result.go

```diff
@@ -59,71 +69,78 @@ func (result *Result) Next(txn Transaction) bool {
 	return true
 }
 
-func (result *Result) getNextPage() error {
+func (result *result) getNextPage() error {
 	nextPage, err := result.communicator.fetchPage(result.ctx, result.pageToken, result.txnID)
 	if err != nil {
 		return err
 	}
+
 	result.pageValues = nextPage.Page.Values
 	result.pageToken = nextPage.Page.NextPageToken
 	result.index = 0
 	result.updateMetrics(nextPage)
 	return nil
 }
 
-func (result *Result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
+func (result *result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
 	if fetchPageResult.ConsumedIOs != nil {
-		*result.metrics.ioUsage.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
-		*result.metrics.ioUsage.writeIOs += *fetchPageResult.ConsumedIOs.WriteIOs
+		*result.ioUsage.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
+		*result.ioUsage.writeIOs += *fetchPageResult.ConsumedIOs.WriteIOs
 	}
 
 	if fetchPageResult.TimingInformation != nil {
-		*result.metrics.timingInformation.processingTimeMilliseconds += *fetchPageResult.TimingInformation.ProcessingTimeMilliseconds
+		*result.timingInfo.processingTimeMilliseconds += *fetchPageResult.TimingInformation.ProcessingTimeMilliseconds
 	}
 }
 
 // GetConsumedIOs returns the statement statistics for the current number of read IO requests that were consumed. The statistics are stateful.
-func (result *Result) GetConsumedIOs() *IOUsage {
-	readIOs := *result.metrics.ioUsage.readIOs
-	writeIOs := *result.metrics.ioUsage.writeIOs
+func (result *result) GetConsumedIOs() *IOUsage {
 	return &IOUsage{
-		readIOs:  &readIOs,
-		writeIOs: &writeIOs,
+		readIOs:  result.ioUsage.readIOs,
+		writeIOs: result.ioUsage.readIOs,
 	}
 }
 
 // GetTimingInformation returns the statement statistics for the current server-side processing time. The statistics are stateful.
-func (result *Result) GetTimingInformation() *TimingInformation {
-	timingInformation := *result.metrics.timingInformation.processingTimeMilliseconds
+func (result *result) GetTimingInformation() *TimingInformation {
 	return &TimingInformation{
-		processingTimeMilliseconds: &timingInformation,
+		processingTimeMilliseconds: result.timingInfo.processingTimeMilliseconds,
 	}
 }
 
 // GetCurrentData returns the current row of data in Ion format. Use ion.Unmarshal or other Ion library methods to handle parsing.
 // See https://github.com/amzn/ion-go for more information.
-func (result *Result) GetCurrentData() []byte {
+func (result *result) GetCurrentData() []byte {
 	return result.ionBinary
 }
 
 // Err returns an error if a previous call to Next has failed.
 // The returned error will be nil if the previous call to Next succeeded.
-func (result *Result) Err() error {
+func (result *result) Err() error {
 	return result.err
 }
 
-// BufferedResult is a cursor over a result set from a QLDB statement that is valid outside the context of a transaction.
```

Comment:
Ditto here regarding the docstring. This is what the user says so it should keep the old docstring.

Reply:
Same question as above.

Reply:
Addressed.